homeassistant/components/switchbot_cloud/sensor.py

```diff
@@ -0,0 +1,119 @@
+"""Platform for sensor integration."""
+
+from __future__ import annotations
+
+from switchbot_api import Device, SwitchBotAPI
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import SwitchbotCloudData
+from .const import DOMAIN
+from .coordinator import SwitchBotCoordinator
+from .entity import SwitchBotCloudEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up SwitchBot Cloud entry."""
+    data: SwitchbotCloudData = hass.data[DOMAIN][config.entry_id]
+
+    entities_to_add = []
+
+    for device, coordinator in data.devices.sensors:
+        entities_to_add.append(
+            _async_make_entity(
+                data.api, device, coordinator, SensorDeviceClass.TEMPERATURE
+            )
+        )
+        entities_to_add.append(
+            _async_make_entity(
+                data.api, device, coordinator, SensorDeviceClass.HUMIDITY
+            )
+        )
+        entities_to_add.append(
+            _async_make_entity(data.api, device, coordinator, SensorDeviceClass.BATTERY)
+        )
+
+    async_add_entities(entities_to_add)
+
+
+class SwitchBotCloudHumiditySensor(
```

Comment:
(continues from last comment)

It would make a lot of sense to use entity descriptions here. That means you create one generic sensor, which is changing shape depending on the entity description you put in.

Reply:
Change made. Code is much cleaner! Thanks!